apps/client/src/pages/UploadPage.vue

```diff
@@ -0,0 +1,149 @@
+<script lang="ts" setup>
+import { ref, computed, watch, onMounted } from 'vue';
+import StepExperimentMetadata from '@/components/upload/StepExperimentMetadata.vue';
+import StepFileSelection from '@/components/upload/StepFileSelection.vue';
+import StepReview from '@/components/upload/StepReview.vue';
+import StepUploadStatus from '@/components/upload/StepUploadStatus.vue';
+import StepColumnNameMapping from '@/components/upload/StepColumnNameMapping.vue';
+import type { QStepper } from 'quasar';
+import { useUploadStore } from '@/stores/uploadStore';
+import { useConfigStore } from '@/stores/configStore';
+import { useGlobalSettings } from '@/stores/globalSettings';
+
+import { router } from '@/router';
+const uploadStore = useUploadStore();
+const configStore = useConfigStore();
+const globalSettings = useGlobalSettings();
+
+// Function to determine if the create experiment button should be enabled.
+function disableUpload(): boolean {
+    return !(
+        experimentMetadataDone() &&
+        fileSelectionDone() &&
+        columnNameMappingDone()
+    );
+}
+
+function experimentMetadataDone(): boolean {
+    return uploadStore.validExperimentName();
+}
+
+function fileSelectionDone(): boolean {
+    return uploadStore.allFilesPopulated() && uploadStore.locationIdsUnique();
+}
+
+function columnNameMappingDone(): boolean {
+    return uploadStore.allColumnsMapped();
+}
+
+function returnHome(): void {
+    router.push('/');
+}
+</script>
+<template>
+    <q-page
+        class="q-pa-lg q-gutter-md"
+        :dark="globalSettings.darkMode"
+        style="max-width: 1200px; margin: auto"
+    >
+        <template v-if="configStore.environment === 'local'">
+            <q-banner inline-actions class="text-white bg-red">
+                The local version of the Loon application does not support data
+                upload directly.
+                <template v-slot:action>
+                    <q-btn
+                        flat
+                        color="white"
+                        label="Return to Home"
+                        @click="returnHome()"
```

Comment:
This makes it a native <a> element (which is preferable) and keeps the button styling. (You also should delete the unused returnHome function)

```suggestion
                        href="/"
```

Reply:
This will cause a full page reload actually and can be less performant. So if you have something that you're viewing and then navigate using href, that view will be lost whereas using router.push it will be maintained.